lib/src/hex_util.rs

```diff
@@ -30,9 +30,15 @@ fn to_forward_hex_digit(b: u8) -> Option<u8> {
 }
 
 pub fn to_forward_hex(reverse_hex: &str) -> Option<String> {
```

Comment:
Can we make the argument a `impl AsRef<[u8]>` and not need two separate functions?